### PR TITLE
[terra-list]  Drag and Drop issue for terra-list in Slide panel 

### PR DESCRIPTION
**Pull request by MadanKumarGovindaswamy**
### Summary

**What was changed:**
Added render clone property for drag and drop context to avoid position loss issue while dragging list item inside the terra slide panel.

**Why it was changed:**
Identified as a bug where the position of dragged or selected items is lost when using Terra List with draggable items inside a Terra Slide Panel component.

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [x] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9620 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-list/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
 ## Unreleased
+* Fixed
```
Comment (supreethmr) [inline packages/terra-list/CHANGELOG.md:4]: *nit : new line for consistency
```suggestion
## Unreleased

* Fixed
```

File: packages/terra-list/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fixed
+  * Fixed DnD issue for `terra-list` inside `terra-slide-panel`.
```
Comment (sdadn) [inline packages/terra-list/CHANGELOG.md:6]: This is too vague. What is the issue? it's also better to spell out "drag and drop" than use an abbreviation.

**Review by sdadn — CHANGES_REQUESTED** [review]
How was the fix tested?

File: packages/terra-list/CHANGELOG.md (hunk 2 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-list/CHANGELOG.md:6]: Identified a bug where the position of dragged or selected items is lost when using Terra List with draggable items inside a Terra Slide Panel component.

**Review by sugan2416 — APPROVED** [review]

**Comment by MadanKumarGovindaswamy** [comment]
> How was the fix tested?

Added a WDIO test case. Thanks

**Review by supreethmr — APPROVED** [review]

**Review by sdadn — APPROVED** [review]

**Pull request merged**